build.gradle

```diff
@@ -52,7 +52,37 @@ forbiddenApisTest.ignoreFailures = true
 validateNebulaPom.enabled = false
 loggerUsageCheck.enabled = false
 
-tasks.matching {it.path in [":publishNebulaPublicationToSnapshotsRepository"]}.all { task ->
+// Order is publish nebula, then jar then zip
```

Comment:
Just for my understanding. Why the order is important here? Because JS publishes JAR and ZIP to different location anyway.

Reply:
This can go in any order of the 3, but down below I had to add explicit task dependencies between generating poms and publishing the artifact associated with the pom. This could also go `jar -> zip -> nebula (spi)` or any other order but I chose this one to get the publishing working.

Reply:
The order is important because we need to manage task to execute by using right group Id. If we do not mention the order then the task is not able to decide which group id it should take it in publishing jars and zip.

For clarity:
The 2 groupId's are
1) org.opensearch
2) org.opensearch.plugin

cc: @cwperks

Reply:
Makes sense. Can we add the detail about the task dependencies of poms and artifacts in the comment to clear the question about the order?

Reply:
Sure.

Reply:
I am assuming this dependency does not block any tasks to be run individually?

Reply:
yes it doesn't

Reply:
@owaiskazi19 added the details of task execution in the description above